### PR TITLE
Add section about Logback dependencies

### PR DESCRIPTION
**Pull request by Emdee89**
## What Has Changed?

Explain in detail which dependencies are needed to use Logback properly.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/guides/logging-overview.mdx [hunk 1]
```diff
@@ -230,6 +230,34 @@ In the case that you want to log all requests sent and responses received by the
 </root>
 ```
 
+#### Required Dependencies 
+
+:::note Only needed if not using the SAP Cloud SDK Maven archetypes
+You only need to add the following dependencies if your project is not based on the SAP Cloud SDK Maven archetypes.
```
Comment (MatKuhr) [inline docs/java/guides/logging-overview.mdx:236]:
```suggestion
You only need to add the following dependencies if your project is not based on one of the [SAP Cloud SDK Maven archetypes](https://search.maven.org/search?q=g:com.sap.cloud.sdk.archetypes).
```

File: docs/java/guides/logging-overview.mdx [hunk 2]
```diff
@@ -230,6 +230,34 @@ In the case that you want to log all requests sent and responses received by the
 </root>
 ```
 
+#### Required Dependencies
```
Comment (MatKuhr) [inline docs/java/guides/logging-overview.mdx:233]: Is this really required by the SDK? Or is it rather required by CF? Also isn't this only required when using Logback? Users might use a different framework...

File: docs/java/guides/logging-overview.mdx [hunk 3]
```diff
@@ -230,6 +230,34 @@ In the case that you want to log all requests sent and responses received by the
...
+#### Required Dependencies 
+
+:::note Only needed if not using the SAP Cloud SDK Maven archetypes
+You only need to add the following dependencies if your project is not based on the SAP Cloud SDK Maven archetypes.
+:::
+
+Add the following dependencies to your project:
+```xml
+<dependency>
+    <groupId>org.slf4j</groupId>
```
Comment (MatKuhr) [inline docs/java/guides/logging-overview.mdx:242]: It's not clear to me why this is required. Isn't this needed only if the user wants to log messages themselves via SLF4J? Users might use Logback directly or any other logging framework.

**Review by MatKuhr — COMMENTED** [review]
The relationship to the SDK is not clear enough to me.

**Comment by Emdee89** [comment]
We do not know enough yet to write proper documentation.

**Pull request closed without merging**